### PR TITLE
Minor cleanup of CFamily helper code

### PR DESCRIPTION
**Pull request by duncanp-sonar**
* extracted path calculations to a separate class
* moved standalone execution method from CFamilyHelper to CLangAnalyzer

### CODE REVIEW

File: src/Core/CFamily/SubProcessFilePaths.cs [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+ */
+
+using System.IO;
+
+namespace SonarLint.VisualStudio.Core.CFamily
+{
+    /// <summary>
+    /// Returns file paths for various CFamily directories/files
+    /// </summary>
+    public static class SubProcessFilePaths
```
Comment (duncanp-sonar) [inline src/Core/CFamily/SubProcessFilePaths.cs:28]: Previously these properties were in `CFamilyHelper` in the `VSIX` project.
I've moved them to this separate class in `Core` so they can be accessed from the new `CFamily` project.

File: src/Integration.Vsix/CFamily/CFamilyHelper.cs [hunk 2]
```diff
@@ -34,20 +32,8 @@ namespace SonarLint.VisualStudio.Integration.Vsix.CFamily
 {
     internal static partial class CFamilyHelper
     {
-        internal static string WorkingDirectory => Path.GetTempPath();
```
Comment (duncanp-sonar) [inline src/Integration.Vsix/CFamily/CFamilyHelper.cs:37]: The properties moved to `Core.CFamily.SubProcessFilesPaths`.

File: src/Integration.Vsix/CFamily/CFamilyHelper.cs [hunk 3]
```diff
@@ -159,55 +145,6 @@ private static void SaveFileConfig(FileConfig fileConfig, ILogger logger)
             return defaults;
         }
 
-        internal /* for testing */ static void CallClangAnalyzer(Action<Message> handleMessage, IRequest request, IProcessRunner runner, ILogger logger, CancellationToken cancellationToken)
```
Comment (duncanp-sonar) [inline src/Integration.Vsix/CFamily/CFamilyHelper.cs:162]: This is the method that actually invoked the sub-process exe. It's a completely standalone method and was only called by the `CLangAnalyzer`, so I've moved it to `CLangAnalyzer`.

That should mean that the only methods left in `CFamilyHelper` are ones that relate to creating the vcxproj `Request` i.e. it should now have a single responsibility. 
I might change this into a non-static class, possibly only that implements some kind of `IRequestBuilder` builder. Either way, the helper class is simpler.

File: src/Integration.Vsix/CFamily/CFamilyHelper.cs [hunk 4]
```diff
@@ -271,38 +208,5 @@ private static Request ToRequest(FileConfig fileConfig, string path)
             }
             return request;
         }
-
-        private static void AddRange(IList<string> cmd, string[] values)
```
Comment (duncanp-sonar) [inline src/Integration.Vsix/CFamily/CFamilyHelper.cs:275]: These methods were only used by the nested `Capture` class, so I've moved them to that class.

File: src/Integration.Vsix/CFamily/CFamilyHelper_Capture.cs [hunk 5]
```diff
@@ -290,6 +290,39 @@ public static Capture[] ToCaptures(FileConfig fileConfig, string path, out strin
                         return "/std:c++14";
                 }
             }
+
```
Comment (duncanp-sonar) [inline src/Integration.Vsix/CFamily/CFamilyHelper_Capture.cs:293]: Moved. No changes.

File: src/Integration.Vsix/CFamily/CLangAnalyzer.cs [hunk 6]
```diff
@@ -189,5 +190,54 @@ private void HandleMessage(Message message, IRequest request, IIssueConsumer con
             // rules will never be active in a quality profile.
             return rulesConfiguration.ActivePartialRuleKeys.Contains(message.RuleKey, CFamilyShared.RuleKeyComparer);
         }
+
+        internal /* for testing */ static void ExecuteSubProcess(Action<Message> handleMessage, IRequest request, IProcessRunner runner, ILogger logger, CancellationToken cancellationToken)
```
Comment (duncanp-sonar) [inline src/Integration.Vsix/CFamily/CLangAnalyzer.cs:194]: Moved from `CFamilyHelper`.

No changes, other than renaming the method and using `SubProcessFilePaths` to get the file paths.

**Comment by duncanp-sonar** [comment]
I've deliberately limited the changes in this PR; I haven't introduced any new interfaces or changed any of the static methods to instance methods. The main changes are as follows:

* the remaining methods in `CFamilyHelper` all relate to create a vcxproj-based `Request`.
* the subprocess-related paths are now provided by a class in `Core`, so they are accessible from the new `CFamily` assembly.

**Review by rita-sonarsource — APPROVED** [review]

**Pull request merged**